Public/Src/Engine/Scheduler/Fingerprints/ObservedInputProcessor.cs

```diff
@@ -1603,20 +1616,20 @@ public Possible<PathExistence> TryProbeAndTrackForExistence(AbsolutePath path, C
                 var fullGraphExistExistence = FileSystemView.GetExistence(path, FileSystemViewMode.FullGraph);
                 if (fullGraphExistExistence.Result == PathExistence.ExistsAsDirectory)
                 {
-                    return PathExistence.ExistsAsDirectory;
+                    existence = PathExistence.ExistsAsDirectory;

```

Comment:
>existence = PathExistence.ExistsAsDirectory; [](start = 20, length = 44)

I prefer these as return statements since it makes it clear that there is no further processing to be done.

Reply:
This is leftover from my debugging -- I tried to check the path existence at the end of the method. I'll revert it.

---
In reply to: [283894583](https://github.com/microsoft/BuildXL/pull/301#discussion_r283894583) [](ancestors = 283894583)